mlrun/runtimes/base.py

```diff
@@ -941,19 +942,32 @@ def _update_run_state(
 
         updates = None
         last_state = get_in(resp, "status.state", "")
+        kind = get_in(resp, "metadata.labels.kind", "")
         if last_state == "error" or err:
-            updates = {"status.last_update": now_date().isoformat()}
-            updates["status.state"] = "error"
+            updates = {
+                "status.last_update": now_date().isoformat(),
+                "status.state": "error",
+            }
             update_in(resp, "status.state", "error")
             if err:
                 update_in(resp, "status.error", err_to_str(err))
             err = get_in(resp, "status.error")
             if err:
                 updates["status.error"] = err_to_str(err)
+
         elif not was_none and last_state != "completed":
-            updates = {"status.last_update": now_date().isoformat()}
-            updates["status.state"] = "completed"
-            update_in(resp, "status.state", "completed")
+
+            # TODO: add a 'workers' section in run objects state, each worker will update its state while
+            #  the run state will be resolved by the server.
+            if kind != "mpijob":
```

Comment:
A question on this - is the situation we saw in mpi only relevant to mpi, or is there some other runtime that may see a similar issue? I assume spark is off the hook since we use the CRD status for its status, what about a job with iterations? Or dask?

Reply:
Dask - mlrun is not running on the workers so we're fine
Spark - I don't know if we run mlrun on the executer / driver it may be a problem
Iterations - each iteration has its own RunObject so we're good
there is also the task_genrator / hyper run which I'm not familiar with and should be considerd